PReLU.lua

```diff
@@ -6,11 +6,11 @@ function PReLU:__init(nOutputPlane)
    self.nOutputPlane = nOutputPlane or 0
    self.weight = torch.Tensor(nOutputPlane or 1):fill(0.25)
    self.gradWeight = torch.Tensor(nOutputPlane or 1)
-   self.gradWeightBuf = torch.Tensor()
-   self.gradWeightBuf2 = torch.Tensor()
 end
 
 function PReLU:updateOutput(input)
+   self.gradWeightBuf = self.gradWeightBuf or input.new()
+   self.gradWeightBuf2 = self.gradWeightBuf2 or input.new()
```

Comment:
2 reasons
* this buffers are needed just because cutorch reduce in not generic enough and cannot reduce with += to a buffer, so should be removed sometime later
* I want to keep all buffers like this out of constructor for backend switching like `cudnn.convert`